Tests/Services/DeviceDetectTest.php

```diff
@@ -66,4 +99,21 @@ public function testCacheManagerDefaultIsOverridable(bool $configLoaded, string
         );
     }
 
+    /**
+     * Test the device detector through multiple user agents.
+     *
+     * @dataProvider userAgentProvider
+     */
+    public function testUserAgentDetector($userAgent, $isDesktop, $isMobile, $isTablet): void
```

Comment:
```suggestion
    /**
     * Test the device detector through multiple user agents.
     *
     * @dataProvider userAgentProvider
     *
     * @param string $userAgent
     * @param bool   $isDesktop
     * @param bool   $isMobile
     * @param bool   $isTablet
     *
     * @return void
     */
    public function testUserAgentDetector(string $userAgent, bool $isDesktop, bool $isMobile, bool $isTablet): void
```
Can we put the types and document the parameters?

Reply:
Ok, changed.